### PR TITLE
Implement influxdb service

### PR DESCRIPTION
**Pull request by serboctor**

### CODE REVIEW

File: pyclient/zeroos/orchestrator/sal/influxdb/templates/__init__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+def get(templatename):
+    templatepath = os.path.dirname(__file__)
+    loader = jinja2.FileSystemLoader(templatepath)
+    env = jinja2.Environment(loader=loader)
+    return env.get_template(templatename)
+
+
+def render(templatename, **kwargs):
+    env = get(templatename)
+    return env.render(**kwargs) + '\n'
```
Comment (grimpy) [inline pyclient/zeroos/orchestrator/sal/influxdb/templates/__init__.py]: i dont like this is duplicated, maybe we should merge the folders somewhere else?

File: templates/influxdb/actions.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    containers = service.producers.get('container')
+    if not containers:
+        raise RuntimeError('Service didn\'t consume any containers')
+
+    j.tools.async.wrappers.sync(containers[0].executeAction('start', context=job.context))
+    container = Container.from_ays(containers[0], job.context['token'])
+    influx = InfluxDB(
+        container, service.parent.model.data.redisAddr, service.model.data.port)
+    influx.start()
+    time.sleep(1)
```
Comment (grimpy) [inline templates/influxdb/actions.py]: start should handle this not a random sleep 1 here

File: templates/influxdb/actions.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    containers = service.producers.get('container')
+    if not containers:
+        raise RuntimeError('Service didn\'t consume any containers')
+
+    container = Container.from_ays(containers[0], get_jwt_token_from_job(job))
+    influx = InfluxDB(
+        container, service.parent.model.data.redisAddr, service.model.data.port)
+
+    if args.get('port'):
```
Comment (grimpy) [inline templates/influxdb/actions.py]: your probably wonna compare if the port changed, if its the same port no need for all this

File: templates/influxdb/actions.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            influx.stop()
+            service.model.data.status = 'halted'
+            influx.port = args['port']
+            influx.start()
+            service.model.data.status = 'running'
+        service.model.data.port = args['port']
+
+    if args.get('databases'):
+        if container.is_running() and influx.is_running()[0]:
+            create_dbs = set(args['databases']) - set(service.model.data.databases)
```
Comment (grimpy) [inline templates/influxdb/actions.py]: you cant compare apple and pears, args['databases'] is a list of dics, data.databases is a capnp list, use model.data.to_dict()['databases'] to compare (see gateway)

File: pyclient/zeroos/orchestrator/sal/influxdb/influxdb.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        while not is_running and time.time() < end:
+            time.sleep(1)
+            is_running, _ = self.is_running()
+
+        if not is_running:
+            self.container.node.client.nft.drop_port(self.port)
+            raise RuntimeError('Failed to start influxd.')
+
+    def create_databases(self, databases):
+        client = j.clients.influxdb.get('172.17.0.1', port=self.port)
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/influxdb/influxdb.py]: opppss

File: templates/influxdb/actions.py (hunk 3 above)
Comment (serboctor) [inline templates/influxdb/actions.py]: It won't be in the args if it is the same value.

File: templates/influxdb/actions.py (hunk 4 above)
Comment (serboctor) [inline templates/influxdb/actions.py]: I've tested it and it works fine.

**Pull request merged**